pyocd/probe/picoprobe.py

```diff
@@ -0,0 +1,636 @@
+    }
+
+    SAFESWD_OPTION = 'picoprobe.safeswd'
+
+    PARITY_BIT = 0x100000000
+
+    @ classmethod
+    def get_all_connected_probes(cls, unique_id=None, is_explicit=False):
+        return [cls(dev) for dev in PicoLink.enumerate_picoprobes()]
+
+    @ classmethod
+    def get_probe_with_id(cls, unique_id, is_explicit=False):
+        for dev in PicoLink.enumerate_picoprobes():
+            if dev.get_unique_id() == unique_id:
+                return cls(dev)
+
+    def __init__(self, picolink):
+        super(Picoprobe, self).__init__()
+        self._link = picolink
+        self._is_connected = False
+        self._is_open = False
+        self._unique_id = self._link.get_unique_id()
+        self._select = -1
+        self._reset = False
+
+    @ property
+    def description(self):
+        return self.vendor_name + " " + self.product_name
+
+    @ property
+    def vendor_name(self):
+        return self._link.vendor_name
+
+    @ property
+    def product_name(self):
+        return self._link.product_name
+
+    @ property
+    def supported_wire_protocols(self):
+        return [DebugProbe.Protocol.DEFAULT, DebugProbe.Protocol.SWD]
+
+    @ property
+    def unique_id(self):
+        return self._unique_id
+
+    @ property
+    def wire_protocol(self):
+        """! @brief Only valid after connecting."""
+        return DebugProbe.Protocol.SWD if self._is_connected else None
+
+    @ property
+    def is_open(self):
+        return self._is_open
+
+    @ property
+    def capabilities(self):
+        return {DebugProbe.Capability.SWJ_SEQUENCE}
+
+    def open(self):
+        self._link.open()
+        self._is_open = True
+        self._select = -1
+
+    def close(self):
+        self._link.close()
+        self._is_open = False
+
+    # ------------------------------------------- #
+    #          Target control functions
+    # ------------------------------------------- #
+    def connect(self, protocol=None):
+        """! @brief Connect to the target via SWD."""
+        # Make sure the protocol is supported
+        if (protocol is None) or (protocol == DebugProbe.Protocol.DEFAULT):
+            protocol = DebugProbe.Protocol.SWD
+
+        # Validate selected protocol.
+        if protocol != DebugProbe.Protocol.SWD:
+            raise ValueError("unsupported wire protocol %s" % protocol)
+
+        self._is_connected = True
+        # Use the bulk or safe read and write functions according to option
+        if self.session.options.get(self.SAFESWD_OPTION):
+            self.read_ap_multiple = self._safe_read_ap_multiple
+            self.write_ap_multiple = self._safe_write_ap_multiple
+        else:
+            self.read_ap_multiple = self._bulk_read_ap_multiple
+            self.write_ap_multiple = self._bulk_write_ap_multiple
+        # Subscribe to option change events
+        self.session.options.subscribe(self._change_options, [self.SAFESWD_OPTION])
+        # Do I need to do anything else here?
```

Comment:
No, nothing else is required.

For most probes, you need to send some command to switch them from Hi-Z to output. This is required so you can have multiple probes connected to a target without conflicts (as long as only one is active at a time, of course).